### PR TITLE
Btfs 1033 payin

### PR DESCRIPTION
**Pull request by ShellyWEI**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior?** (You can also refer to a JIRA ticket here)

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **What dependencies / modules need to be updated as pre-requisites?** (Include relevant PR links here)

* **Description of changes**

---

* **Please check if the PR fulfills these requirements**
- [x] The subject of this PR contains a JIRA ticket BTFS-xxxx (N/A for community PRs)
- [ ] PR description and commit messages are [good and meaningful](https://chris.beams.io/posts/git-commit/)
- [ ] Unit tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **Please make sure the following procedures have been applied before requesting reviewers**
- [ ] Code changes closely follow [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
- [ ] Code has been *rebased* against latest master (`git merge` not recommended, unless you know what you are doing)
- [ ] Code changes have run through `go fmt`
- [ ] Code changes have run through `go mod tidy`
- [ ] All unit tests passed locally (`make test_go_test`)

### CODE REVIEW

File: core/escrow/escrow.go [hunk 1]
```diff
@@ -67,33 +68,79 @@ func NewContractRequest(configuration *config.Config, signedContracts []*escrowp
...
+	payerAddr, err := payerPubKey.Raw()
+	if err != nil {
+		return nil, err
+	}
+	payinReq := &escrowpb.PayinRquest{
+		PayinId:           result.Result.PayinId,
+		BuyerAddress:      payerAddr,
+		BuyerChannelState: chanState,
+	}
+	payinSig, err := crypto.Sign(payerPrivKey, payinReq)
```
Comment (taiyangc) [inline core/escrow/escrow.go]: return err

File: core/escrow/escrow.go [hunk 2]
```diff
@@ -67,33 +68,79 @@ func NewContractRequest(configuration *config.Config, signedContracts []*escrowp
...
+	return &escrowpb.SignedPayinRquest{
+		Request:        payinReq,
+		BuyerSignature: payinSig,
+	}, nil
+}
+
+func PayInToEscrow(configuration *config.Config, signedPayinReq *escrowpb.SignedPayinRquest) error {
+	conn, err := grpc.Dial("52.15.101.94:50051", grpc.WithInsecure())
```
Comment (taiyangc) [inline core/escrow/escrow.go]: @laipogo please help Shelly with this grpc conn since this is a new path of code

File: core/escrow/escrow.go [hunk 3]
```diff
@@ -67,33 +68,79 @@ func NewContractRequest(configuration *config.Config, signedContracts []*escrowp
 	}, nil
 }
 
-func SubmitContractToEscrow(configuration *config.Config, request *escrowpb.EscrowContractRequest) error {
+func SubmitContractToEscrow(configuration *config.Config, request *escrowpb.EscrowContractRequest) (*escrowpb.SignedSubmitContractResult, error) {
 	var conn *grpc.ClientConn
 	// TODO: Make escrow IP hidden in config too, now for testing purpose leave it here
 	conn, err := grpc.Dial("52.15.101.94:50051", grpc.WithInsecure())
```
Comment (taiyangc) [inline core/escrow/escrow.go]: similarly, this needs an update

File: core/escrow/escrow.go (hunk 2 above)
Comment (laipogo) [reply core/escrow/escrow.go]: > @laipogo please help Shelly with this grpc conn since this is a new path of code

ok, will do

**Comment by taiyangc** [comment]
@laipogo conflicts with master

File: core/escrow/escrow.go [hunk 4]
```diff
@@ -67,33 +68,82 @@ func NewContractRequest(configuration *config.Config, signedContracts []*escrowp
...
-	if err != nil {
-		return err
-	}
-	if response == nil {
-		return fmt.Errorf("escrow reponse is nil")
-	}
+func SubmitContractToEscrow(configuration *config.Config, request *escrowpb.EscrowContractRequest) (
+	response *escrowpb.SignedSubmitContractResult, err error) {
+	//TODO: for testing purpose
+	address := "52.15.101.94:50051"
```
Comment (taiyangc) [inline core/escrow/escrow.go]: ready from `configuration.Services.EscrowDomain` please. For testing, please change your local `~/.btfs/config`value for this field to `http://52.15.101.94:50051`

Comment (taiyangc) [reply core/escrow/escrow.go]: (otherwise the testbed won't work well)

Comment (laipogo) [reply core/escrow/escrow.go]: oh, i see. that's great we can override it somewhere.

**Pull request merged**